### PR TITLE
[Core][Geometry] BrepCurveOnSurface restructurement

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

I am changing here the way that data is structured within the BrepCurveOnSurface. Now the NurbsCurveOnSurface class is used instead of both independently.

Thx!

### CODE REVIEW

File: kratos/geometries/brep_curve_on_surface.h [hunk 1]
```diff
@@ -172,6 +207,21 @@ class BrepCurveOnSurface
         return typename BaseType::Pointer( new BrepCurveOnSurface( ThisPoints ) );
     }
 
+    ///@}
+    ///@name Set/ Get functions
+    ///@{
+
+    bool GetCurveDirection()
+    {
+        return mCurveDirection;
```
Comment (philbucher) [inline kratos/geometries/brep_curve_on_surface.h]: maybe here sth like you use in #6215 would make sense here too

**Review by philbucher — COMMENTED** [review]
@pooyan-dadvand any comments here?

File: kratos/geometries/brep_curve_on_surface.h (hunk 1 above)
Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h]: It is something coming from the input. And allows only, same direction or controverse direction.

Comment (pooyan-dadvand) [reply kratos/geometries/brep_curve_on_surface.h]: For me at least the name of the method is confusing and the return value deosn't help. What is the meaning of ture or false?

Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h]: Didn't you find the comment declarative enough? ;)

I changed it to more convenient name mSameCurveDirection and added some comments.

Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h]: @philbucher do you agree with the new name? Is it more understandable?

File: kratos/geometries/brep_curve_on_surface.h [hunk 2]
```diff
@@ -172,6 +211,27 @@ class BrepCurveOnSurface
...
+    * @return true -> brep curve and nurbs curve point in same direction.
+    *        false -> brep curve and nurbs curve point in controversial directions.
+    */
+    bool HasSameCurveDirection()
+    {
+        return mSameCurveDirection;
+    }
+
+    /// Returns the curve on surface of this brep.
+    NurbsCurveOnSurfacePointerType pGetCurveOnSurface()
```
Comment (philbucher) [inline kratos/geometries/brep_curve_on_surface.h:230]: you are missing the const version

File: kratos/geometries/brep_curve_on_surface.h [hunk 3]
```diff
@@ -282,11 +328,14 @@ class BrepCurveOnSurface
...
     ///@{
 
-    typename NurbsSurfaceType::Pointer mpNurbsSurface;
-    typename NurbsCurveType::Pointer mpNurbsCurve;
+    NurbsCurveOnSurfacePointerType mpCurveOnSurface;
 
     NurbsInterval mCurveNurbsInterval;
 
+    /** true-> brep curve and nurbs curve point in same direction.
+    *  false-> brep curve and nurbs curve point in controversial directions. */
```
Comment (philbucher) [inline kratos/geometries/brep_curve_on_surface.h]:
```suggestion
    *  false-> brep curve and nurbs curve point in opposite directions. */
```

**Review by philbucher — COMMENTED** [review]
one comment otherwise looks good to me :+1:

File: kratos/geometries/brep_curve_on_surface.h (hunk 2 above)
Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h:230]: Thanks, sure. It is here now ;)

**Comment by tteschemacher** [comment]
Thanks for your comments. I tried to address them!

**Review by pooyan-dadvand — APPROVED** [review]
Better than before!

**Comment by tteschemacher** [comment]
Thanks!

**Pull request merged**